File: leetcode/c++/lc274.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int hIndex(vector<int>& citations) {
        int n = (int)citations.size();
        sort(citations.begin(),citations.end());
        if(citations[0] >= n) return n;
        if(n == 1) return !(citations[0] == 0);
        for(int i = 0; i < n; i++) {
            int left = n - i;
            if(citations[i] >= left) return left;
        }
        return 0;        
    }
};

int main() {
    vector<int> citations = {1,1,3};
    Solution s;
    cout << s.hIndex(citations) << endl;
    return 0;
}

File: leetcode/python/lc274.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        h = 0
        n = len(citations)
        citations.sort(reverse=True)
        for i in range(n):
            if citations[i] >= i + 1:
                h = i + 1
        return h


citations = [3, 0, 6, 1, 5]
s = Solution()
print(s.hIndex(citations))
